### PR TITLE
fix: update treesitter functions

### PR DESCRIPTION
**Pull request by danilshvalov**
See neovim/neovim#22761 (maybe we should wait for the 0.9 release)

### CODE REVIEW

**Comment by jgollenz** [comment]
> maybe we should wait for the 0.9 release

That depends on whether we want to ... um ... depend on it :grin: Sooner or later we will, but I let @kristijanhusak decide that.

**Review by mcepl — APPROVED** [review]
Works for me (and fixes https://github.com/nvim-orgmode/orgmode/issues/536 for me).

**Comment by jgollenz** [comment]
@danilshvalov do you mind squashing your commits? I'm not sure why the tests are not re-running after I changed the target branch, but a force push from you should do the trick.

**Comment by kristijanhusak** [comment]
We are usually trying to support two latest versions + nightly, which will be 0.8.3, 0.9 and nightly, once the 0.9 is released. I'm ok with this, but we should probably make some wrapper around these and use appropriate one depending on the Neovim version.

**Comment by jgollenz** [comment]
@danilshvalov in this case you can set the base branch back to `master`

**Comment by jgollenz** [comment]
On `0.8.3` I also need a shim for `set_query` (used in `todo_highlighter.lua`) and use `ts.get_query` in `markup_highlighter.lua` and `todo_highlighter.lua` in order to not get errors.
On `nightly` I get a notice about deprecation of `vim.treesitter.query.get_query()` upon every startup. Can we avoid that?

**Review by jgollenz — DISMISSED** [review]
- On `0.8.[0|3]` I also need 
	- a shim for `set_query` (used in `todo_highlighter.lua:55`) 
	- use `ts.get_qery` in `markup_highlighter.lua:230` 
	- use `ts.get_query_files` in `todo_highlighter.lua:8`

  in order to not get errors.
- On `nightly` I get a notice about deprecation of `vim.treesitter.query.get_query()` upon every startup. Can we avoid that?

**Comment by danilshvalov** [comment]
> * On `0.8.[0|3]` I also need
>   
>   * a shim for `set_query` (used in `todo_highlighter.lua:55`)
>   * use `ts.get_qery` in `markup_highlighter.lua:230`
>   * use `ts.get_query_files` in `todo_highlighter.lua:8`

Done

>   in order to not get errors.
> * On `nightly` I get a notice about deprecation of `vim.treesitter.query.get_query()` upon every startup. Can we avoid that?

I can't reproduce it. Could you provide a log please?

**Comment by danilshvalov** [comment]
@jgollenz, is there anything else?

File: lua/orgmode/org/indent.lua [hunk 1]
```diff
@@ -25,7 +25,7 @@ local get_matches = ts_utils.memoize_by_buf_tick(function(bufnr)
       }
 
       if type == 'headline' then
-        opts.stars = vim.treesitter.query.get_node_text(node:field('stars')[1], bufnr):len()
+        opts.stars = vim.treesitter.get_node_text(node:field('stars')[1], bufnr):len()
```
Comment (kristijanhusak) [inline lua/orgmode/org/indent.lua:28]: I think we need to use compat here and below in whole file

Comment (danilshvalov) [reply lua/orgmode/org/indent.lua]: Done

**Comment by jgollenz** [comment]
@danilshvalov when I run the `minimal_init` with your branch I don't get the error, so it looks like it's just on my side. I'll investigate further, but let's merge this in the meantime :+1:

**Pull request merged**